### DISCUSSION TITLE
Possible unreachable code in routing.py

### DISCUSSION THREAD

**Discussion by b4sen** [fastapi/fastapi#5958] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
if prefix:
    assert prefix.startswith("/"), "A path prefix must start with '/'"
    assert not prefix.endswith(
        "/"
    ), "A path prefix must not end with '/', as the routes will start with '/'"
else:
    for r in router.routes:
        path = getattr(r, "path")  # noqa: B009
        name = getattr(r, "name", "unknown")
        if path is not None and not path:
            raise Exception(
                f"Prefix and path cannot be both empty (path operation: {name})"
            )
```

### Description

The `if path is not None and not path:` in [this line](https://github.com/tiangolo/fastapi/blob/40df42f5c7d92815ceeb546df60256dacb3eed57/fastapi/routing.py#L714) will be always falsy thus never raising the exception. I think it should be `if path is None:`

### Operating System

macOS

### FastAPI Version

current master

### Python Version

3.10

**Comment by jgould22** [comment, marked as answer]
```python
In [1]: path = ""

In [2]: path is not None and not path
Out[2]: True
```
`is` checks if it is the same object `and` checks if it is falsy

**Reply by b4sen** [reply]
True, thank you! I first thought "and" calls bool() implicitly but didn't check the default value of path.
